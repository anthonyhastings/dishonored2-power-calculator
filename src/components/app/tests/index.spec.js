import noop from 'lodash/noop';
import renderer from 'react-test-renderer';
import App from '../';

jest.mock('react-router-dom', () => ({
  Switch: 'MockSwitch',
  Route: 'MockRoute',
  Link: 'MockLink',
}));

jest.mock(
  'Components/character-selection/container',
  () =>
    function MockCharacterSelection() {
      return <mock-character-selection />;
    }
);

jest.mock('Components/loader', () => 'MockLoader');

jest.mock(
  'Components/page-not-found',
  () =>
    function MockPageNotFound() {
      return <mock-page-not-found />;
    }
);

jest.mock(
  'Components/power-selection/container/route-validation',
  () =>
    function MockPowerSelectionRouteValidation() {
      return <mock-power-selection-route-validation />;
    }
);

describe('App component', () => {
  let testContext;

  const renderComponent = (props) => (
    <App {...testContext.defaultProps} {...props} />
  );

  beforeEach(() => {
    testContext = {};

    testContext.defaultProps = {
      onComponentDidMount: noop,
      showError: false,
      showLoader: true,
    };
  });

  describe('when rendered', () => {
    beforeEach(async () => {
      testContext.onComponentDidMountMock = jest.fn();

      await renderer.act(async () => {
        testContext.component = renderer.create(
          renderComponent({
            onComponentDidMount: testContext.onComponentDidMountMock,
          })
        );
      });
    });

    it('renders a header area and top-level routes', () => {
      expect(testContext.component).toMatchSnapshot();
    });

    it('triggers prop callback on mounting', () => {
      expect(testContext.onComponentDidMountMock).toHaveBeenCalledTimes(1);
    });
  });
});
