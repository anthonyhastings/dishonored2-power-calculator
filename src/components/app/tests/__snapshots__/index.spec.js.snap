// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component when rendered renders a header area and top-level routes 1`] = `
<div
  className="app"
>
  <header
    className="app__header"
  >
    <h1
      className="app__title"
    >
      Power Calculator
    </h1>
    <MockLink
      className="app__logo-link"
      to="/"
    >
      <img
        alt="Dishonored 2 logo"
        className="app__logo-graphic"
        sizes="100vw"
        srcSet="660x90.png 660w, 1320x180.png 1320w"
      />
    </MockLink>
  </header>
  <div
    className="app__content"
  >
    <MockLoader
      showError={false}
      showLoader={true}
    >
      <MockSwitch>
        <MockRoute
          exact={true}
          path="/"
        >
          <mock-character-selection />
        </MockRoute>
        <MockRoute
          path="/powers/:characterSlug"
        >
          <mock-power-selection-route-validation />
        </MockRoute>
        <MockRoute>
          <mock-page-not-found />
        </MockRoute>
      </MockSwitch>
    </MockLoader>
  </div>
</div>
`;
